pkg/api/v20170131/types.go

```diff
@@ -109,6 +110,23 @@ type MasterProfile struct {
 	subnet string
 }
 
+// UnmarshalJSON unmarshal json using the default behavior
+// And do fields manipulation, such as populating default value
+func (m *MasterProfile) UnmarshalJSON(b []byte) error {
+	// Need to have a alias type to avoid circular unmarshal
+	type aliasMasterProfile MasterProfile
```

Comment:
otherwise, when you call MasterProfile.UnmarshalJSON, it causes a circular call to itself.